spec/models/manageiq/providers/openshift/container_manager/refresh_parser_spec.rb

```diff
@@ -281,40 +281,99 @@ def parse_single_openshift_image_with_registry
-                                   :phase                         => 'set to stun',
-                                   :reason                        => 'this is a reason',
-                                   :duration                      => '33',
-                                   :completion_timestamp          => '50',
-                                   :start_timestamp               => '17',
-                                   :labels                        => [],
-                                   :build_config                  => nil,
-                                   :output_docker_image_reference => 'host:port/path/to/image'
-                                  )
+                         RecursiveOpenStruct.new(build_pod)
+                         )).to eq(:name                          => 'ruby-sample-build-1',
+                                  :ems_ref                       => 'af3d1a10-44c0-11e5-b186-0aaeec44370e',
+                                  :namespace                     => 'test-namespace',
+                                  :ems_created_on                => '2015-08-17T09:16:46Z',
+                                  :resource_version              => '165339',
+                                  :message                       => 'we come in peace',
+                                  :phase                         => 'set to stun',
+                                  :reason                        => 'this is a reason',
+                                  :duration                      => '33',
+                                  :completion_timestamp          => '50',
+                                  :start_timestamp               => '17',
+                                  :labels                        => [],
+                                  :build_config                  => nil,
+                                  :output_docker_image_reference => 'host:port/path/to/image'
+                                 )
+    end
+
+    context "build config and pods linking" do
+      def parse_entities(namespace_pod, namespace_config)
+        build_pod = basic_build_pod.dup
+        build_pod[:metadata][:namespace] = namespace_pod
+        build_pod[:status][:config][:namespace] = namespace_pod
+        build_config = basic_build_config.dup
+        build_config[:metadata][:namespace] = namespace_config
+        parser.get_builds(RecursiveOpenStruct.new({"build_config" => [RecursiveOpenStruct.new(build_config),]}))
+        parser.get_build_pods(RecursiveOpenStruct.new({"build" => [RecursiveOpenStruct.new(build_pod),]}))
+      end
+
+      it "links correct build pods to build configurations in same namespace" do
+        parse_entities('namespace_1', 'namespace_1')
+        expect(parser.instance_variable_get('@data')[parser.send(:path_for_entity, "build")].first[:build_config]).to eq(
+          parser.instance_variable_get('@data')[parser.send(:path_for_entity, "build_config")].first
```

Comment:
I'd hardcode the results of `path_for_entity`, both for readability and because I prefer concreteness in tests.
LGTM either way.